entity-service-client/src/main/java/org/hypertrace/entity/data/service/client/EdsCacheClient.java

```diff
@@ -224,4 +226,72 @@ public void upsertEnrichedEntities(String tenantId, EnrichedEntities enrichedEnt
   public void upsertRelationships(String tenantId, EntityRelationships relationships) {
     client.upsertRelationships(tenantId, relationships);
   }
+
+  public void updateBasedOnChangeEvent(
+      EntityChangeEventKey entityChangeEventKey, EntityChangeEventValue entityChangeEventValue) {
+    LOG.debug("Entity change event is {}, {} ", entityChangeEventKey, entityChangeEventValue);
+    Entity entity;
+
+    switch (entityChangeEventValue.getEventCase()) {
+      case CREATE_EVENT:
+        // ignore create events, don't populate caches if not necessary
+        break;
+      case UPDATE_EVENT:
+        entity = entityChangeEventValue.getUpdateEvent().getLatestEntity();
+        updateCacheValues(entityChangeEventKey, entity);
+        break;
+      case DELETE_EVENT:
+        entity = entityChangeEventValue.getDeleteEvent().getDeletedEntity();
+        entityCache.invalidate(getEntityCacheKeyWithoutEntityType(entityChangeEventKey));
```

Comment:
doesn't this have to invalidate the one with entity type too? It seems like it would make sense to share the code that iterates the various cache entries that can be associated with an entity.